### ISSUE TITLE
Encrypt self-sent messages?

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-core#104]
The upcoming version of Delta Chat (0.14.0 then) will have an option to send self-sent messages. The idea is that you can eg. send a photo to this chat and have the photo then on your other devices, eg. Thunderbird, a Tablet with K-9 or whatever.

You'll find similar functions in some other messengers as "Cloud" or "Saved messages", however, eg. WhatsApp or Signal do not have such a feature at all as they have no real multi-device support at all.

**The question I would like to discuss here,** is if the messages sent to the self-chat should be encrypted or not.

- by default, Delta Chat prefers encryption by setting [prefer-encrypt=mutual](https://autocrypt.org/level1.html#the-autocrypt-header)
- if the user uses eg. Delta Chat and eg. GMX-Web, and sends messages from Delta Chat to another Delta Chat user, he won't be able to read these messages in GMX-Web, however, as the messages are moved away to the "Chats" folder, the user might not expect to see the mails there (in general, you can send to key to another device, but this is another topic)
- for self-sent messages, however, this might not be acceptable - if a message is sent from Delta Chat to the self-chat, the user might expect to read just this message on all of his devices, also eg. in the GMX-Webfrontend

**Possible solutions** may be 
- not to encrypt to this self-chat by default - and add an option to do so
- not to default to prefer-encrypt=mutual at all - the user has to set up all devices properly and enable the encryption manually then (I won't like this, i still see delta as a chance to raise the number of overall email-encryption)

However, these are just _my_ ideas, there may be much better solutions - what do you think? 

Btw: I just used a self-sent message to get a screenshot from the phone to the desktop ...

![self-sent-chat](https://user-images.githubusercontent.com/9800740/36422060-14cdb636-163b-11e8-8693-a3aa292ce80c.png)

... this is how the self-sent chat looks like

Still btw: The whole self-sent chat function is only a side effect of the just implemented setup messages, see #35

**Comment by hpk42** [comment]
On Tue, Feb 20, 2018 at 11:53 +0000, Björn Petersen wrote:
> The upcoming version of Delta Chat (0.14.0 then) will have an option to send self-sent messages. The idea is that you can eg. send a photo to this chat and have the photo then on your other devices, eg. Thunderbird, a Tablet with K-9 or whatever.

So this is a special chat "me", right? 

> **Possible solutions** may be 
> - not to encrypt to this self-chat by default - and add an option to do so
> - not to default to prefer-encrypt=mutual at all - the user has to set up all devices properly and enable the encryption manually then (I won't like this, i still see delta as a chance to raise the number of overall email-encryption)

i think option 2 is not good, it would disable the current "delta.chat will e2e
encrypt between users" behaviour which is a big feature. 

> However, these are just _my_ ideas, there may be much better solutions - what do you think? 

Given that there are desktop options (thunderbird+enigmail) to read/write encrypted
mail with autocrypt, and as delta/enigmail both support Autocrypt-setup soon, 
i guess it's ok to make no exception for self-sent messages, i.e. encrypt them. 
This renders them unreadable by webmail but hopefully soon mailvelope/gmx will
provide a way to have the autocrypt key available through the web. 

What would help practically, is if a chat can be manually forced to cleartext. 
So if a user is in the situation to want to have self-sent messages readable
everywhere irrespective of synced autocrypt support, then it could be switched
on for that chat.

**Comment by r10s** [comment]
> So this is a special chat "me", right?

Yes. The "me" chat is not visible by default, one has to create it like any other chat. It is automatically created when receiving a self-sent message - as an Autocrypt Setup Message.

> i think option 2 is not good, it would disable the current "delta.chat will e2e encrypt between users" behaviour which is a big feature

I agree.

> Given that there are desktop options (thunderbird+enigmail) to read/write encrypted mail with autocrypt, and as delta/enigmail both support Autocrypt-setup soon, i guess it's ok to make no exception for self-sent messages, i.e. encrypt them. This renders them unreadable by webmail but hopefully soon mailvelope/gmx will provide a way to have the autocrypt key available through the web.

I have added this as a 3rd option above. Maybe this is the easiest solution is the best solution in this case :)

> What would help practically, is if a chat can be manually forced to cleartext. So if a user is in the situation to want to have self-sent messages readable everywhere irrespective of synced autocrypt support, then it could be switched on for that chat.

Yes, maybe an option for this would be handy, see also https://github.com/deltachat/deltachat-android/issues/246

**Comment by testbird** [comment]
My 2ct. would agree with leaving the encrypt-by-default on, and have a force(-once) override. But how in the UI long press on the send button may be "dangerous". Maybe, long press on the lock symbol in the message to write? (-> posting this to the other issue)

**Comment by testbird** [comment]
Forgot to say, awesome feature! :)
